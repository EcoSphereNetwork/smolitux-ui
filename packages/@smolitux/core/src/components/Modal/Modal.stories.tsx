import type { Meta, StoryObj } from '@storybook/react';
import { Modal } from './Modal';

const meta: Meta<typeof Modal> = {
  title: 'Components/Modal',
  component: Modal,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Modal',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom Modal',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive Modal',
    onClick: () => alert('Clicked!'),
  },
};
