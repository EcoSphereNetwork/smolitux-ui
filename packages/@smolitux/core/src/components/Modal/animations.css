/* Fade-In Animation */
@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.animate-fadeIn {
  animation: fadeIn 0.3s ease-in-out forwards;
}

/* Scale-In Animation */
@keyframes scaleIn {
  from {
    opacity: 0;
    transform: scale(0.95);
  }
  to {
    opacity: 1;
    transform: scale(1);
  }
}

.animate-scaleIn {
  animation: scaleIn 0.3s ease-in-out forwards;
}

/* Slide-In-Up Animation */
@keyframes slideInUp {
  from {
    opacity: 0;
    transform: translateY(20px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

.animate-slideInUp {
  animation: slideInUp 0.3s ease-in-out forwards;
}

/* Slide-In-Down Animation */
@keyframes slideInDown {
  from {
    opacity: 0;
    transform: translateY(-20px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

.animate-slideInDown {
  animation: slideInDown 0.3s ease-in-out forwards;
}

/* Slide-In-Left Animation */
@keyframes slideInLeft {
  from {
    opacity: 0;
    transform: translateX(-20px);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
}

.animate-slideInLeft {
  animation: slideInLeft 0.3s ease-in-out forwards;
}

/* Slide-In-Right Animation */
@keyframes slideInRight {
  from {
    opacity: 0;
    transform: translateX(20px);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
}

.animate-slideInRight {
  animation: slideInRight 0.3s ease-in-out forwards;
}