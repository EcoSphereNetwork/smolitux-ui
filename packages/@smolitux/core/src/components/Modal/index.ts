export { Modal } from './Modal';
export type { ModalProps } from './Modal';

// Für Abwärtskompatibilität mit dem bestehenden Export
export { Modal as default } from './Modal';