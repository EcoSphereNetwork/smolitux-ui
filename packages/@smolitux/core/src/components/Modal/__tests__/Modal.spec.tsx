import React from 'react';
import renderer from 'react-test-renderer';
import { Modal } from '../Modal';

// Mock fÃ¼r createPortal
jest.mock('react-dom', () => {
  const originalModule = jest.requireActual('react-dom');
  return {
    ...originalModule,
    createPortal: (node: React.ReactNode) => node,
  };
});

describe('Modal Snapshots', () => {
  test('renders correctly with default props', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}}>
        <p>Modal Content</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with title', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} title="Test Modal">
        <p>Modal Content</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with footer', () => {
    const tree = renderer.create(
      <Modal 
        isOpen={true} 
        onClose={() => {}} 
        title="Test Modal"
        footer={<button>Save</button>}
      >
        <p>Modal Content</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with small size', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} size="sm">
        <p>Small Modal</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with large size', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} size="lg">
        <p>Large Modal</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with top position', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} position="top">
        <p>Top Modal</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with bottom position', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} position="bottom">
        <p>Bottom Modal</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly without close button', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} showCloseButton={false}>
        <p>Modal without close button</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom className', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} className="custom-class">
        <p>Modal with custom class</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom ID', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} id="custom-modal-id">
        <p>Modal with custom ID</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly without animation', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} animated={false}>
        <p>Modal without animation</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly without shadow', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} shadow={false}>
        <p>Modal without shadow</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly without rounded corners', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} rounded={false}>
        <p>Modal without rounded corners</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly without border', () => {
    const tree = renderer.create(
      <Modal isOpen={true} onClose={() => {}} bordered={false}>
        <p>Modal without border</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with multiple props combined', () => {
    const tree = renderer.create(
      <Modal 
        isOpen={true} 
        onClose={() => {}} 
        title="Combined Modal"
        footer={<button>Save</button>}
        size="lg"
        position="top"
        className="custom-class"
        headerClassName="header-class"
        bodyClassName="body-class"
        footerClassName="footer-class"
        id="combined-modal"
        animated={false}
        showCloseButton={true}
        shadow={true}
        rounded={true}
        bordered={true}
        scrollable={true}
      >
        <p>Modal with multiple props</p>
      </Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});