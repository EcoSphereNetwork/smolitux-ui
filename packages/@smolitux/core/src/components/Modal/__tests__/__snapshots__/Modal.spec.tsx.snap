// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snapshots renders correctly with bottom position 1`] = `
<div
  aria-describedby="modal-8mg0r52uk-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-end justify-center pb-10"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700  "
      id="modal-8mg0r52uk"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-8mg0r52uk-body"
      >
        <p>
          Bottom Modal
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with custom ID 1`] = `
<div
  aria-describedby="custom-modal-id-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity animate-fadeIn "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700 animate-scaleIn "
      id="custom-modal-id"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="custom-modal-id-body"
      >
        <p>
          Modal with custom ID
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with custom className 1`] = `
<div
  aria-describedby="modal-szcf4i1gj-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700  custom-class"
      id="modal-szcf4i1gj"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-szcf4i1gj-body"
      >
        <p>
          Modal with custom class
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with default props 1`] = `
<div
  aria-describedby="modal-9wisssv33-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700  "
      id="modal-9wisssv33"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-9wisssv33-body"
      >
        <p>
          Modal Content
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with footer 1`] = `
<div
  aria-describedby="modal-q1atbrt1h-body"
  aria-labelledby="modal-q1atbrt1h-title"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700  "
      id="modal-q1atbrt1h"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <h3
          className="text-lg font-medium text-gray-900 dark:text-white"
          id="modal-q1atbrt1h-title"
        >
          Test Modal
        </h3>
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-q1atbrt1h-body"
      >
        <p>
          Modal Content
        </p>
      </div>
      <div
        className="border-t border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button>
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with large size 1`] = `
<div
  aria-describedby="modal-i778wzrql-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-lg shadow-xl rounded-lg border border-gray-200 dark:border-gray-700  "
      id="modal-i778wzrql"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-i778wzrql-body"
      >
        <p>
          Large Modal
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with multiple props combined 1`] = `
<div
  aria-describedby="combined-modal-body"
  aria-labelledby="combined-modal-title"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-start justify-center pt-10"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-lg shadow-xl rounded-lg border border-gray-200 dark:border-gray-700  custom-class"
      id="combined-modal"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 header-class"
      >
        <h3
          className="text-lg font-medium text-gray-900 dark:text-white"
          id="combined-modal-title"
        >
          Combined Modal
        </h3>
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 body-class"
        id="combined-modal-body"
      >
        <p>
          Modal with multiple props
        </p>
      </div>
      <div
        className="border-t border-gray-200 dark:border-gray-700 px-6 py-4 footer-class"
      >
        <button>
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with small size 1`] = `
<div
  aria-describedby="modal-wgd8alp91-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity animate-fadeIn "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-sm shadow-xl rounded-lg border border-gray-200 dark:border-gray-700 animate-scaleIn "
      id="modal-wgd8alp91"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-wgd8alp91-body"
      >
        <p>
          Small Modal
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with title 1`] = `
<div
  aria-describedby="modal-pr1s6120n-body"
  aria-labelledby="modal-pr1s6120n-title"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity animate-fadeIn "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700 animate-scaleIn "
      id="modal-pr1s6120n"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <h3
          className="text-lg font-medium text-gray-900 dark:text-white"
          id="modal-pr1s6120n-title"
        >
          Test Modal
        </h3>
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-pr1s6120n-body"
      >
        <p>
          Modal Content
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly with top position 1`] = `
<div
  aria-describedby="modal-a6fusfagx-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity animate-fadeIn "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-start justify-center pt-10"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700 animate-scaleIn "
      id="modal-a6fusfagx"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-a6fusfagx-body"
      >
        <p>
          Top Modal
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly without animation 1`] = `
<div
  aria-describedby="modal-fqm1bhkvq-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700  "
      id="modal-fqm1bhkvq"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-fqm1bhkvq-body"
      >
        <p>
          Modal without animation
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly without border 1`] = `
<div
  aria-describedby="modal-c0g87eyys-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg   "
      id="modal-c0g87eyys"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-c0g87eyys-body"
      >
        <p>
          Modal without border
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly without close button 1`] = `
<div
  aria-describedby="modal-0olszxfvz-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity animate-fadeIn "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl rounded-lg border border-gray-200 dark:border-gray-700 animate-scaleIn "
      id="modal-0olszxfvz"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="p-6 "
        id="modal-0olszxfvz-body"
      >
        <p>
          Modal without close button
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly without rounded corners 1`] = `
<div
  aria-describedby="modal-zajsp1ibx-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity animate-fadeIn "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md shadow-xl  border border-gray-200 dark:border-gray-700 animate-scaleIn "
      id="modal-zajsp1ibx"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-zajsp1ibx-body"
      >
        <p>
          Modal without rounded corners
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal Snapshots renders correctly without shadow 1`] = `
<div
  aria-describedby="modal-d0vsks2i2-body"
  aria-modal="true"
  className="fixed inset-0 z-50 overflow-y-auto "
  role="dialog"
>
  <div
    aria-hidden="true"
    className="fixed inset-0 bg-black bg-opacity-50 transition-opacity  "
    onClick={[Function]}
  />
  <div
    className="flex min-h-screen p-4 items-center justify-center"
  >
    <div
      className="relative bg-white dark:bg-gray-800 text-left transform transition-all w-full max-w-md  rounded-lg border border-gray-200 dark:border-gray-700  "
      id="modal-d0vsks2i2"
      onClick={[Function]}
      tabIndex={-1}
    >
      <div
        className="border-b border-gray-200 dark:border-gray-700 px-6 py-4 "
      >
        <button
          aria-label="Close"
          className="absolute top-3 right-3 text-gray-400 hover:text-gray-500 dark:hover:text-gray-300 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            className="h-6 w-6"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="p-6 "
        id="modal-d0vsks2i2-body"
      >
        <p>
          Modal without shadow
        </p>
      </div>
    </div>
  </div>
</div>
`;
